### PR TITLE
HIP-584: Acceptance tests for eth_call with precompile logic

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

 Acceptance tests using Cucumber, validating that eth_call with token precompile support works as expected

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java [hunk 1]
```diff
@@ -207,6 +209,48 @@ public NetworkTransactionResponse approveCryptoAllowance(AccountId spender, Hbar
...
+    public NetworkTransactionResponse approveNft(NftId nftId, AccountId spender) {
+        var ownerAccountId = sdkClient.getExpandedOperatorAccountId().getAccountId();
+
+        var transaction = new AccountAllowanceApproveTransaction()
+                .approveTokenNftAllowance(nftId, ownerAccountId, spender);
+
+        NetworkTransactionResponse networkTransactionResponse =
+                executeTransactionAndRetrieveReceipt(transaction);
+
+        log.debug("Sent Account Allowance Approval");
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java:221]: Probably we can be more specific for the type of approval.

`log.debug("Sent Account Allowance Approval")` -> `log.debug("Sent Account Allowance Approval for Nft with id: { } and serialNumber: { }", nftId.getId(), nftId.getSerialNumber())`

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java [hunk 2]
```diff
@@ -207,6 +209,48 @@ public NetworkTransactionResponse approveCryptoAllowance(AccountId spender, Hbar
...
+    public NetworkTransactionResponse approveToken(TokenId tokenId, AccountId spender, long amount) {
+        var ownerAccountId = sdkClient.getExpandedOperatorAccountId().getAccountId();
+
+        var transaction = new AccountAllowanceApproveTransaction()
+                .approveTokenAllowance(tokenId, ownerAccountId, spender, amount);
+
+        NetworkTransactionResponse networkTransactionResponse =
+                executeTransactionAndRetrieveReceipt(transaction);
+
+        log.debug("Sent Account Allowance Approval");
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java:235]: `log.debug("Sent Account Allowance Approval") -> log.debug("Sent Account Allowance Approval for Token with id: { } and amount: { }", tokenId, amount))`

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java [hunk 3]
```diff
@@ -207,6 +209,48 @@ public NetworkTransactionResponse approveCryptoAllowance(AccountId spender, Hbar
...
+    public NetworkTransactionResponse approveNftAllSerials(TokenId tokenId, AccountId spender) {
+        var ownerAccountId = sdkClient.getExpandedOperatorAccountId().getAccountId();
+
+        var transaction = new AccountAllowanceApproveTransaction()
+                .approveTokenNftAllowanceAllSerials(tokenId, ownerAccountId, spender);
+
+        NetworkTransactionResponse networkTransactionResponse =
+                executeTransactionAndRetrieveReceipt(transaction);
+
+        log.debug("Sent Account Allowance Approval");
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/AccountClient.java:249]: `log.debug("Sent Account Allowance Approval") -> log.debug("Sent Account Allowance Approval for all serial numbers for Nft with id: { }, tokenId)`

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+        allowanceSpenderAccountId = accountFeature.setTokenAllowance(accountName, tokenIds.get(0), amount);
+    }
+
+    @Then("I approve {string} for nft all serials")
+    public void approveCryptoAllowanceAllSerials(String accountName) {
+        spenderAccountIdForAllSeerials = accountFeature.setNftAllowanceAllSerials(accountName, tokenIds.get(1));
+    }
+
+
+    public void createNewToken(String symbol, int freezeStatus, int kycStatus) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java:334]: Probably rename to `createNewFungibleToken`

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+    public void createNewContract() throws IOException {
+        compiledSolidityArtifact = MAPPER.readValue(
+                ResourceUtils.getFile(ercContract.toUri()),
+                CompiledSolidityArtifact.class);
+        contractId = contractFeature.createContract(compiledSolidityArtifact.getBytecode(), 0);
+    }
+
+    @Then("I create a new token with freeze status 2 and kyc status 1")
+    public void createNewFungibleToken() {
+        createNewFungibleToken("TEST", TokenFreezeStatus.FreezeNotApplicable_VALUE, TokenKycStatus.KycNotApplicable_VALUE);
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java:291]: remove these methods

- `public void createNewFungibleToken(String symbol, int freezeStatus, int kycStatus)`
- `private void createNewFungibleToken(String symbol, int freezeStatus, int kycStatus, TokenType tokenType,
            TokenSupplyType tokenSupplyType, List<CustomFee> customFees)`

and just implement the logic in  this method.

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java [hunk 6]
```diff
@@ -508,12 +508,12 @@ private TokenId createNewToken(String symbol, int freezeStatus, int kycStatus, T
         return tokenId;
     }
 
-    private void createNewToken(String symbol, int freezeStatus, int kycStatus) {
-        createNewToken(symbol, freezeStatus, kycStatus, TokenType.FUNGIBLE_COMMON, TokenSupplyType.INFINITE, Collections
+    public TokenId createNewToken(String symbol, int freezeStatus, int kycStatus) {
+        return createNewToken(symbol, freezeStatus, kycStatus, TokenType.FUNGIBLE_COMMON, TokenSupplyType.INFINITE, Collections
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java:512]: revert

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java [hunk 7]
```diff
@@ -508,12 +508,12 @@ private TokenId createNewToken(String symbol, int freezeStatus, int kycStatus, T
...
                 .emptyList());
     }
 
-    private void createNewNft(String symbol, int freezeStatus, int kycStatus, TokenSupplyType tokenSupplyType) {
+    public TokenId createNewNft(String symbol, int freezeStatus, int kycStatus, TokenSupplyType tokenSupplyType) {
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java:516]: common code should go to TokenClient, it's bad practice to provide it as a public method and call it in another feature class

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+    public void approveTokenAllowance(String accountName, long amount) {
+        allowanceSpenderAccountId = accountFeature.setTokenAllowance(accountName, tokenIds.get(0), amount);
+    }
+
+    @Then("I approve {string} for nft all serials")
+    public void approveCryptoAllowanceAllSerials(String accountName) {
+        spenderAccountIdForAllSeerials = accountFeature.setNftAllowanceAllSerials(accountName, tokenIds.get(1));
+    }
+
+
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/ERCContractFeature.java:333]: need an `@After` method to clean up resources

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/response/ContractCallResponse.java [hunk 9]
```diff
@@ -39,4 +42,14 @@ public static String convertContractCallResponseToSelector(final ContractCallRes
     public static String convertContractCallResponseToAddress(final ContractCallResponse response) {
         return Bytes.fromHexString(response.getResult()).slice(12).toUnprefixedHexString();
     }
+
+    public static String hexToASCII(String str) throws DecoderException {
```
Comment (steven-sheehy) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/response/ContractCallResponse.java:46]: Would prefer these two new methods as non-static methods.

```suggestion
    public String getResultAsText() throws DecoderException {
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/response/ContractCallResponse.java [hunk 10]
```diff
@@ -39,4 +44,13 @@ public static String convertContractCallResponseToSelector(final ContractCallRes
     public static String convertContractCallResponseToAddress(final ContractCallResponse response) {
         return Bytes.fromHexString(response.getResult()).slice(12).toUnprefixedHexString();
     }
+
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/response/ContractCallResponse.java:47]: just noticed all methods in this class (including the static methods) have the response itself as the parameter. we should make static methods non static and just access the `result` field inside without passing anything in

Comment (xin-hedera) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/response/ContractCallResponse.java:47]: sounds good then, can leave it to a separate PR

**Review by xin-hedera — DISMISSED** [review]
LGTM

**Comment by steven-sheehy** [comment]
Testing this against integration which has the token precompile code and getting a contract revert for token owner. Can you please check?
```
Then I mint a serial number                                                                            # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.mintNftToken()
2023-03-28T21:41:40.031-0600 INFO reactor-http-nio-3 c.h.m.t.e.a.c.MirrorNodeClient POST https://localhost/api/v1/contracts/call: 200 OK
And I call the erc contract via the mirror node REST API for token name                                # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.nameContractCall()
2023-03-28T21:41:40.072-0600 INFO reactor-http-nio-2 c.h.m.t.e.a.c.MirrorNodeClient GET https://localhost/api/v1/transactions/0.0.1353-1680061287-395724708: 200 OK
Then the mirror node REST API should return status 200 for the crypto transfer transaction   # com.hedera.mirror.test.e2e.acceptance.steps.AccountFeature.verifyMirrorAPICryptoTransferResponse(int)
2023-03-28T21:41:40.095-0600 INFO reactor-http-nio-4 c.h.m.t.e.a.c.MirrorNodeClient POST https://localhost/api/v1/contracts/call: 200 OK
And I call the erc contract via the mirror node REST API for token symbol                              # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.symbolContractCall()
2023-03-28T21:41:40.136-0600 WARN ForkJoinPool-3-worker-3 c.h.h.s.TransactionReceiptQuery Retrying in 250 ms after failure with node 0.0.3 during attempt #1: OK
2023-03-28T21:41:40.235-0600 INFO reactor-http-nio-3 c.h.m.t.e.a.c.MirrorNodeClient POST https://localhost/api/v1/contracts/call: 200 OK
And I call the erc contract via the mirror node REST API for token decimals                            # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.decimalsContractCall()
2023-03-28T21:41:40.291-0600 INFO reactor-http-nio-2 c.h.m.t.e.a.c.MirrorNodeClient POST https://localhost/api/v1/contracts/call: 200 OK
And I call the erc contract via the mirror node REST API for token totalSupply                         # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.totalSupplyContractCall()
2023-03-28T21:41:40.317-0600 INFO reactor-http-nio-4 c.h.m.t.e.a.c.MirrorNodeClient GET https://localhost/api/v1/accounts/CIQM5HVLXY5BGAVCNJS3CWS4DJ62VVL2NYXHWHVVJEHXQRZBF45ZTSQ: 404 NOT_FOUND
2023-03-28T21:41:40.341-0600 WARN ForkJoinPool-3-worker-1 c.h.h.s.TransactionReceiptQuery Retrying in 250 ms after failure with node 0.0.6 during attempt #1: RECEIPT_NOT_FOUND
2023-03-28T21:41:40.414-0600 WARN ForkJoinPool-3-worker-3 c.h.h.s.TransactionReceiptQuery Retrying in 500 ms after failure with node 0.0.5 during attempt #2: RECEIPT_NOT_FOUND
2023-03-28T21:41:40.423-0600 INFO reactor-http-nio-3 c.h.m.t.e.a.c.MirrorNodeClient POST https://localhost/api/v1/contracts/call: 400 BAD_REQUEST
2023-03-28T21:41:40.425-0600 ERROR reactor-http-nio-3 c.h.m.t.e.a.c.MirrorNodeClient Endpoint failed, returning: 400 Bad Request from POST https://localhost/api/v1/contracts/call
And I call the erc contract via the mirror node REST API for token ownerOf                             # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.ownerOfContractCall()
org.springframework.web.reactive.function.client.WebClientResponseException$BadRequest: 400 Bad Request from POST https://localhost/api/v1/contracts/call
at org.springframework.web.reactive.function.client.WebClientResponseException.create(WebClientResponseException.java:217)
at ✽.I call the erc contract via the mirror node REST API for token ownerOf(classpath:features/contract/erc.feature:14)
Suppressed: The stacktrace has been enhanced by Reactor, refer to additional information below:
Error has been observed at the following site(s):
*__checkpoint ⇢ 400 from POST https://localhost/api/v1/contracts/call [DefaultWebClient]
Original Stack Trace:
at org.springframework.web.reactive.function.client.WebClientResponseException.create(WebClientResponseException.java:217)
at org.springframework.web.reactive.function.client.DefaultClientResponse.lambda$createException$1(DefaultClientResponse.java:207)
at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106)
at reactor.core.publisher.FluxOnErrorReturn$ReturnSubscriber.onNext(FluxOnErrorReturn.java:162)
at reactor.core.publisher.FluxDefaultIfEmpty$DefaultIfEmptySubscriber.onNext(FluxDefaultIfEmpty.java:101)
at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:129)
at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107)
at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:299)
at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:337)
at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1839)
at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:160)
at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:144)
at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:260)
at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:144)
at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:415)
at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:424)
at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:478)
at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:712)
at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:113)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1382)
at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1245)
at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1294)
at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
at java.base/java.lang.Thread.run(Thread.java:833)
Suppressed: java.lang.Exception: #block terminated with an error
at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99)
at reactor.core.publisher.Mono.block(Mono.java:1742)
at com.hedera.mirror.test.e2e.acceptance.client.MirrorNodeClient.callPostRestEndpoint(MirrorNodeClient.java:277)
at com.hedera.mirror.test.e2e.acceptance.client.MirrorNodeClient.contractsCall(MirrorNodeClient.java:183)
at com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.ownerOfContractCall(ERCContractFeature.java:171)
at com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature$$FastClassBySpringCGLIB$$7b376075.invoke(<generated>)
at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:97)
at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:133)
at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:159)
at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
at com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature$$EnhancerBySpringCGLIB$$2a2cef3e.ownerOfContractCall(<generated>)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at io.cucumber.java.Invoker.doInvoke(Invoker.java:66)
at io.cucumber.java.Invoker.invoke(Invoker.java:24)
at io.cucumber.java.AbstractGlueDefinition.invokeMethod(AbstractGlueDefinition.java:47)
at io.cucumber.java.JavaStepDefinition.execute(JavaStepDefinition.java:29)
at io.cucumber.core.runner.CoreStepDefinition.execute(CoreStepDefinition.java:66)
at io.cucumber.core.runner.PickleStepDefinitionMatch.runStep(PickleStepDefinitionMatch.java:63)
at io.cucumber.core.runner.ExecutionMode$1.execute(ExecutionMode.java:10)
at io.cucumber.core.runner.TestStep.executeStep(TestStep.java:84)
at io.cucumber.core.runner.TestStep.run(TestStep.java:56)
at io.cucumber.core.runner.PickleStepTestStep.run(PickleStepTestStep.java:51)
at io.cucumber.core.runner.TestCase.run(TestCase.java:84)
at io.cucumber.core.runner.Runner.runPickle(Runner.java:75)
at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.lambda$runTestCase$4(CucumberEngineExecutionContext.java:112)
at io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:130)
at io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)
at io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:130)
at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.runTestCase(CucumberEngineExecutionContext.java:109)
at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:164)
at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:90)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

And I call the erc contract via the mirror node REST API for token tokenUri                            # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.tokenURIContractCall()
And I call the erc contract via the mirror node REST API for token getApproved                         # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.getApprovedContractCall()
And I call the erc contract via the mirror node REST API for token allowance                           # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.allowanceContractCall()
And I call the erc contract via the mirror node REST API for token isApprovedForAll                    # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.isApprovedForAllContractCall()
And I call the erc contract via the mirror node REST API for token balance                             # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.balanceOfContractCall()
Then I approve "BOB" for nft                                                                           # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.approveCryptoAllowance(java.lang.String)
And I call the erc contract via the mirror node REST API for token getApproved with response BOB       # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.verifyNftAllowance()
Then I approve "ALICE" for nft all serials                                                             # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.approveCryptoAllowanceAllSerials(java.lang.String)
And I call the erc contract via the mirror node REST API for token isApprovedForAll with response true # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.isApprovedForAllSecondContractCall()
Then I approve "ALICE" with 2                                                                          # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.approveTokenAllowance(java.lang.String,long)
And I call the erc contract via the mirror node REST API for token allowance with allowances           # com.hedera.mirror.test.e2e.acceptance.steps.ERCContractFeature.allowanceSecondContractCall()
2023-03-28T21:41:40.438-0600 INFO reactor-http-nio-2 c.h.m.t.e.a.c.MirrorNodeClient GET https://localhost/api/v1/transactions/0.0.1353-1680061288-781913517: 200 OK
```

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**